contracts/mechs/nevermined/BalanceTrackerNvmSubscriptionNative.sol

```diff
@@ -79,31 +79,25 @@ contract BalanceTrackerNvmSubscriptionNative is BalanceTrackerFixedPriceNative {
         // Get requester actual subscription balance
         uint256 subscriptionBalance = IERC1155(subscriptionNFT).balanceOf(requester, subscriptionTokenId);
 
-        // Adjust requester balance with maxDeliveryRate credits
-        balance += maxDeliveryRate;
+        uint256 totalBalance = balance + subscriptionBalance;
 
         // Check the request delivery rate for a fixed price
-        if (subscriptionBalance < balance) {
-            revert InsufficientBalance(subscriptionBalance, balance);
+        if (totalBalance < maxDeliveryRate) {
+            revert InsufficientBalance(totalBalance, maxDeliveryRate);
         }
 
-        return balance;
-    }
-
-    /// @dev Adjusts final requester balance accounting for possible delivery rate difference (credit).
-    /// @param requester Requester address.
-    /// @param rateDiff Delivery rate difference.
-    /// @return Adjusted balance.
-    function _adjustFinalBalance(address requester, uint256 rateDiff) internal virtual override returns (uint256) {
-        uint256 balance = mapRequesterBalances[requester];
-
-        // This must never happen as max delivery rate is always bigger or equal to the actual delivery rate
-        if (rateDiff > balance) {
-            revert Overflow(rateDiff, balance);
+        // Calculate how many actual credits need to burn accounting for discount balance
+        if (balance > maxDeliveryRate) {
+            balance -= maxDeliveryRate;
+        } else {
+            maxDeliveryRate -= balance;
+            balance = 0;
+            IERC1155(subscriptionNFT).burn(requester, subscriptionTokenId, maxDeliveryRate);
         }
 
-        // Adjust requester credit balance
-        return (balance - rateDiff);
+        emit RequesterCreditsRedeemed(requester, maxDeliveryRate);
```

Comment:
Although this also makes sense in the previous iteration this even would have been emitted in the `else` block right after `burn`.

Reply:
Thanks, fixed in #101